grpc/src/test/java/com/linecorp/armeria/server/grpc/GrpcServiceServerTest.java

```diff
@@ -17,98 +17,927 @@
+        @SuppressWarnings("unchecked")
+        StreamObserver<StreamingOutputCallResponse> responseObserver =
+                mock(StreamObserver.class);
+        StreamObserver<StreamingOutputCallRequest> requestObserver
+                = asyncStub.fullDuplexCall(responseObserver);
+        requestObserver.onNext(streamingRequest);
+        requestObserver.onCompleted();
+
+        ArgumentCaptor<Throwable> captor = ArgumentCaptor.forClass(Throwable.class);
+        verify(responseObserver, timeout(operationTimeoutMillis())).onError(captor.capture());
+        assertThat(Status.fromThrowable(captor.getValue()).getCode()).isEqualTo(Status.UNKNOWN.getCode());
+        assertThat(Status.fromThrowable(captor.getValue()).getDescription()).isEqualTo(errorMessage);
+        verifyNoMoreInteractions(responseObserver);
+    }
+
+    /** Sends an rpc to an unimplemented method within TestService. */
+    @Test(timeout = 10000)
+    public void unimplementedMethod() {
+        Throwable t = catchThrowable(() -> blockingStub.unimplementedCall(Empty.getDefaultInstance()));
+        assertThat(t).isInstanceOf(StatusRuntimeException.class);
+        assertThat(((StatusRuntimeException) t).getStatus().getCode())
+                .isEqualTo(Status.UNIMPLEMENTED.getCode());
+    }
+
+    @Test(timeout = 10000)
+    public void unimplementedService() {
+        UnimplementedServiceGrpc.UnimplementedServiceBlockingStub stub =
+                UnimplementedServiceGrpc.newBlockingStub(asyncStub.getChannel());
+        Throwable t = catchThrowable(() -> stub.unimplementedCall(Empty.getDefaultInstance()));
+        assertThat(t).isInstanceOf(StatusRuntimeException.class);
+        assertThat(((StatusRuntimeException) t).getStatus().getCode())
+                .isEqualTo(Status.UNIMPLEMENTED.getCode());
+    }
+
+    /** Start a fullDuplexCall which the server will not respond, and verify the deadline expires. */
+    @Test(timeout = 10000)
+    public void timeoutOnSleepingServer() throws Exception {
+        TestServiceGrpc.TestServiceStub stub = Clients.newDerivedClient(
+                asyncStub,
+                (ClientBuilderParams) asyncStub.getChannel(),
+                ClientOption.DEFAULT_RESPONSE_TIMEOUT_MILLIS.newValue(1L));
+
+        StreamRecorder<StreamingOutputCallResponse> responseObserver = StreamRecorder.create();
+        StreamObserver<StreamingOutputCallRequest> requestObserver
+                = stub.fullDuplexCall(responseObserver);
+
+        StreamingOutputCallRequest request =
+                StreamingOutputCallRequest.newBuilder()
+                                          .setPayload(Payload.newBuilder()
+                                                             .setBody(ByteString.copyFrom(new byte[27182])))
+                                          .build();
+        try {
+            requestObserver.onNext(request);
+            // TODO(anuraag): Upstream test does not need to call onCompleted - response timeout is set at the
```

Comment:
Found that the response timeout is only set after req.close(). This doesn't work with streaming where we'd expect the response timeout to be active during a streaming request.